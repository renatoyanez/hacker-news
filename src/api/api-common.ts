import axios from 'axios';

export default axios.create({
	baseURL: 'https://hn.algolia.com/api/v1/',
	headers: {
		'Content-type': 'application/json',
	},
});
