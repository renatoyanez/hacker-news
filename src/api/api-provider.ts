import API from './api-common';
import { INews } from 'interfaces';

export const useApi = () => {
	return {
		news: (query: string, page: number): Promise<INews> => {
			return API.get('/search_by_date', {
				params: { query, page },
			});
		},
	};
};
