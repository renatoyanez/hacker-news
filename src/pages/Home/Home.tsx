import { useState, ChangeEvent } from 'react';
import { Dropdown, NewsListSection } from 'components';
import { useGetNews } from 'hooks';
import angular from 'assets/images/angular-logo.jpg';
import vue from 'assets/images/vue-logo.jpg';
import react from 'assets/images/react-logo.jpg';
import styles from './home.module.css';

const Home = () => {
	const [placeholder] = useState('Select your news');

	const options = [
		{ id: 2, name: 'Angular', image: angular, value: 'Angular' },
		{ id: 3, name: 'React', image: react, value: 'React' },
		{ id: 4, name: 'Vue', image: vue, value: 'Vue' },
	];

	const [value, setValue] = useState('');

	const handleChange = (event: ChangeEvent<HTMLSelectElement>) => {
		setValue(event.target.value);
	};
	const { news, error, isLoading } = useGetNews({
		query: 'reactjs',
		page: 3,
	});
	// console.log(value);

	return (
		<div className={styles.home}>
			<Dropdown
				options={options}
				value={value}
				placeholder={placeholder}
				onChange={handleChange}
			/>
			<NewsListSection />
		</div>
	);
};

export default Home;
