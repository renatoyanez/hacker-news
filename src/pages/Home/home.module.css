@import '../../styles/variables.css';

.home {
	background-color: var(--white);
}
