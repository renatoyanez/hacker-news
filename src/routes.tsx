import { ReactElement } from 'react';
import { Route, Routes as Switch } from 'react-router-dom';
import { Faves, Home } from 'pages';

type RouteListType = {
	HOME: string;
	FAVES: string;
};

export const RouteList: RouteListType = {
	HOME: '/',
	FAVES: '/my-faves',
};

const Routes = (): ReactElement => {
	return (
		<Switch>
			<Route path={RouteList.HOME} element={<Home />} />
			<Route path={RouteList.FAVES} element={<Faves />} />
		</Switch>
	);
};

export default Routes;
