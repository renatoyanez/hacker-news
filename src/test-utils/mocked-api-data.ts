import { IHits, INews } from 'interfaces';

export function getMockedApiDataPaginated(): INews {
	return {
		data: {
			hits: [] as IHits[],
			page: 1,
			nbPages: 1,
			hitsPerPage: 1,
			query: 'string',
		},
	};
}
