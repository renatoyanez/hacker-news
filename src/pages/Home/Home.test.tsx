import React from 'react';
import { Router as MemRouter } from 'react-router-dom';
import { useApi } from '../../api/api-provider';
import { getMockedApiDataPaginated } from 'test-utils/mocked-api-data';
import { render, cleanup } from '@testing-library/react';
import { create } from 'react-test-renderer';
import { act } from 'react-dom/test-utils';
import { createMemoryHistory } from 'history';
import { RouteList } from 'routes';
import Home from './Home';

jest.mock('api/api-provider');
afterEach(cleanup);

describe('Home', () => {
	const promise = Promise.resolve();

	jest.spyOn(Object.getPrototypeOf(window.localStorage), 'setItem');
	Object.setPrototypeOf(window.localStorage.setItem, jest.fn());

	const history = createMemoryHistory({
		initialEntries: [RouteList.HOME],
	});

	it('renders correctly', () => {
		const tree = create(
			<MemRouter location={history.location} navigator={history}>
				<Home />
			</MemRouter>,
		).toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('sets data to the local storage', async () => {
		jest.mock('../../api/api-provider');
		jest.mocked(useApi).mockImplementation(() => {
			return {
				news: () =>
					new Promise((resolve, _reject) =>
						resolve(getMockedApiDataPaginated()),
					),
			};
		});
		act(() => {
			render(
				<MemRouter location={history.location} navigator={history}>
					<Home />
				</MemRouter>,
			);
		});
		expect(window.localStorage.setItem).toHaveBeenCalledWith(
			'filter',
			'',
		);
		await act(() => promise);
	});
});
