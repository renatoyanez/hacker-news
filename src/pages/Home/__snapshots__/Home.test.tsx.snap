// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders correctly 1`] = `
<div
  className="home"
>
  <div>
    <div
      className="container"
    >
      <div
        className="selectedText"
        data-testid="dropdown-component"
        onClick={[Function]}
      >
        <p
          className="textStyle4"
        >
          Select your news
        </p>
      </div>
    </div>
  </div>
  <div
    style={
      Object {
        "marginBottom": "2.375rem",
      }
    }
  />
  <div>
    Loading...
  </div>
</div>
`;
